types/types.go

```diff
@@ -47,9 +47,12 @@ type ImageDestination interface {
 // Image is a Docker image in a repository.
 type Image interface {
 	// ref to repository?
+	// GetManifest is like ImageSource.GetManifest, but the result is cached; it is OK to call this however often you need.
+	GetManifest() ([]byte, error)
+	// GetSignatures is like ImageSource.GetSignatures, but the result is cached; it is OK to call this however often you need.
+	GetSignatures() ([][]byte, error)
```

Comment:
I do think that the source/destination distinction is useful; but for sources, this PR already goes a long way towards most users using a `types.Image` and not caring about `ImageSource` at all (basically unless you need layer contents, in which case tarball streams and ordering etc. are really important, `Image` gives you the same information and the caching and parsing makes it the preferred interface.).